controller/tracker.go

```diff
@@ -63,96 +76,150 @@ func (c *TrackerController) Create(ctx *app.CreateTrackerContext) error {
+	}
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		tracker, err := appl.Trackers().Load(ctx.Context, id)
 		if err != nil {
-			cause := errs.Cause(err)
-			switch cause.(type) {
-			case remoteworkitem.NotFoundError:
-				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrNotFound(err.Error()))
-				return ctx.NotFound(jerrors)
-			default:
-				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInternal(err.Error()))
-				return ctx.InternalServerError(jerrors)
-			}
+			return err
 		}
-		return ctx.OK([]byte{})
+		return appl.Trackers().Delete(ctx.Context, tracker.ID)
 	})
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
 	accessTokens := GetAccessTokens(c.configuration) //configuration.GetGithubAuthToken()
 	c.scheduler.ScheduleAllQueries(ctx, accessTokens)
-	return result
+	return ctx.OK([]byte{})
 }
 
 // Show runs the show action.
 func (c *TrackerController) Show(ctx *app.ShowTrackerContext) error {
+	id, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+	}
 	return application.Transactional(c.db, func(appl application.Application) error {
-		t, err := appl.Trackers().Load(ctx.Context, ctx.ID)
+		tracker, err := appl.Trackers().Load(ctx.Context, id)
 		if err != nil {
-			cause := errs.Cause(err)
-			switch cause.(type) {
-			case remoteworkitem.NotFoundError:
-				log.Error(ctx, map[string]interface{}{
-					"tracker_id": ctx.ID,
-				}, "tracker controller not found")
-				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrNotFound(err.Error()))
-				return ctx.NotFound(jerrors)
-			default:
-				jerrors, httpStatusCode := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInternal(err.Error()))
-				return ctx.ResponseData.Service.Send(ctx.Context, httpStatusCode, jerrors)
-			}
+			log.Error(ctx, map[string]interface{}{
+				"err":        err,
+				"tracker_id": ctx.ID,
+			}, "unable to load the tracker by ID")
+			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-		return ctx.OK(t)
+		result := &app.TrackerSingle{
+			Data: convertTracker(appl, ctx.Request, *tracker),
+		}
+		return ctx.OK(result)
 	})
 }
 
 // List runs the list action.
 func (c *TrackerController) List(ctx *app.ListTrackerContext) error {
-	exp, err := query.Parse(ctx.Filter)
-	if err != nil {
-		jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrBadRequest(fmt.Sprintf("could not parse filter: %s", err.Error())))
-		return ctx.BadRequest(jerrors)
-	}
-	start, limit, err := parseLimit(ctx.Page)
-	if err != nil {
-		jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrBadRequest(fmt.Sprintf("could not parse paging: %s", err.Error())))
-		return ctx.BadRequest(jerrors)
-	}
 	return application.Transactional(c.db, func(appl application.Application) error {
-		result, err := appl.Trackers().List(ctx.Context, exp, start, &limit)
+		trackers, err := appl.Trackers().List(ctx)
```

Comment:
So, trackers don't need pagination?

Reply:
@baijum It's a one time import from remote issue tracker.